opm/simulators/utils/PropsCentroidsDataHandle.hpp

```diff
@@ -101,15 +110,26 @@ class PropsCentroidsDataHandle
                 data.reserve(m_no_data);
 
                 for (const auto& intKey : m_intKeys)
-                    data.push_back(globalProps.get_int(intKey)[index]);
+                {
+                    const auto& fieldData = globalProps.get_int_field_data(intKey);
+                    data.push_back(std::make_pair(fieldData.data[index],
+                                                  static_cast<unsigned char>(fieldData.value_status[index])));
+                }
 
                 for (const auto& doubleKey : m_doubleKeys)
-                    data.push_back(globalProps.get_double(doubleKey)[index]);
+                {
+                    // We need to allow unsupported keywords to get the data
+                    // for TranCalculator, too.
+                    const auto& fieldData = globalProps.get_double_field_data(doubleKey,
+                                                                              /* allow_unsupported = */ true);
+                    data.push_back(std::make_pair(fieldData.data[index],
+                                                  static_cast<unsigned char>(fieldData.value_status[index])));
+                }
 
                 auto cartIndex = cartMapper.cartesianIndex(index);
                 const auto& center = eclGridOnRoot->getCellCenter(cartIndex);
                 for (int dim = 0; dim < Grid::dimensionworld; ++dim)
-                    data.push_back(center[dim]);
+                    data.push_back(std::make_pair(center[dim], '1')); // write garbage for value_status
```

Comment:
I guess you could do:
```
#include <opm/parser/eclipse/Deck/value_status.hpp>
...
...
data.push_back(std::make_pair(center[dim], static_cast<int>( value::status::uninitialized )));
```

Reply:
Nope. Wrong cast. Yes, I could do a lot of things here, but it just might not be worth ti. Please be aware that this value is read and thrown away, anyway.

Reply:
Tangentially, why do you use `push_back(make_pair())` when `emplace_back()` will do the same thing?

Reply:
Because I am old school and tend to use what I know (and was really unsure whether pair isn't an aggregate. Do you really think a temporary is a performance penalty here?

Feel free to change, later

Reply:
Changed